openlibrary/plugins/upstream/account.py

```diff
@@ -692,8 +695,12 @@ def GET(self):
 
     @require_login
     def POST(self):
+        i = web.input(public_readlog="", safe_mode="")
         user = accounts.get_current_user()
```

Comment:
After 699, before we save, we want to fetch the user's **current** preferences (prior to saving) to see if safe_mode is already `yes` on the account.

We will increment stats by one for `ol.account.safemode` if and only if the patron's current preferences don't include safe_mode yet.